### PR TITLE
GSK-2822 Don't edit global logger level

### PR DESCRIPTION
**Pull request by kevinmessiaen**
## Description

Updated logger to only update giskard log level to INFO by default

## Related Issue

- [GSK-2822 (Available on Linear)](https://linear.app/giskard/issue/GSK-2822/dont-edit-global-logger-level)

## Type of Change

- [ ] 📚 Examples / docs / tutorials / dependencies update
- [x] 🔧 Bug fix (non-breaking change which fixes an issue)
- [ ] 🥂 Improvement (non-breaking change which improves an existing feature)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔐 Security fix

### CODE REVIEW

File: tests/utils/test_logging_utils.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+
+def test_other_package_log_level_unset():
+    assert (
+        logging.getLogger(openai.__name__).level == logging.NOTSET
+    ), "Non giskard package log level should't be touched by giskard (NOTSET)"
+
+
+def test_root_log_level_default_warning():
+    assert logging.root.level == logging.WARNING, 'Root package log level should be set to default "WARNING"'
```
Comment (Hartorn) [inline tests/utils/test_logging_utils.py:20]: I'm all for this modification, but for scan and worker, we should have some logs, no ?

Comment (kevinmessiaen) [reply tests/utils/test_logging_utils.py:20]: Yes you're right, it should be an arg for the CLI to allow flexibility

File: giskard/commands/cli_worker.py [hunk 2]
```diff
@@ -104,8 +118,11 @@ def start_command(url: AnyHttpUrl, is_server, api_key, is_daemon, hf_token, nb_w
     """
     analytics.track(
         "giskard-worker:start",
-        {"is_server": is_server, "url": anonymize(url), "is_daemon": is_daemon},
+        {"is_server": is_server, "url": anonymize(url), "is_daemon": is_daemon, "log_level": log_level},
     )
+
+    logging.root.setLevel(logging.getLevelName(log_level))
```
Comment (Hartorn) [inline giskard/commands/cli_worker.py:124]: Setting log level does not work for me : if I set it to error, I still have logs from giskard weirdly. Not sure why

Comment (Hartorn) [reply giskard/commands/cli_worker.py:124]: Normally setting Error level to root level should suppress every log even from giskard, in my case worker is still starting with logs.

Comment (Hartorn) [reply giskard/commands/cli_worker.py:124]: Nvm, it was only at start, seems ok after

**Review by Hartorn — APPROVED** [review]

**Pull request merged**